python/IAM_USER_MATCHES_REGEX_PATTERN/IAM_USER_MATCHES_REGEX_PATTERN_test.py

```diff
@@ -0,0 +1,295 @@
+        response = RULE.lambda_handler(lambda_event, {})
+        print(response)
+        assert_customer_error_response(self, response, "InvalidParameterValueException")
+
+    def test_scenario_4_non_compliant(self):
+        """Test scenario to test non-compliant user name
+        Keyword arguments:
+        self -- class ComplianceTest
+        """
+        IAM_CLIENT_MOCK.list_users = MagicMock(return_value=self.user_list)
+        rule_param = '{"pattern":"admin"}'
+        invoking_event = constructInvokingEvent(constructConfigItem(self.user_list['Users'][0]['UserName']))
+        lambda_event = build_lambda_configurationchange_event(invoking_event, rule_parameters=rule_param)
+        response = RULE.lambda_handler(lambda_event, {})
+        print(response)
+        resp_expected = []
+        resp_expected.append(build_expected_response('NON_COMPLIANT', self.user_list['Users'][0]['UserName'], annotation='The regex (admin) does not match (user-name-1).'))
+        assert_successful_evaluation(self, response, resp_expected, 1)
+
+    def test_scenario_5_compliant(self):
+        """Test scenario to test compliant user name
+        Keyword arguments:
+        self -- class ComplianceTest
+        """
+        IAM_CLIENT_MOCK.list_users = MagicMock(return_value=self.user_list)
+        rule_param = '{"pattern":".*admin.*"}'
+        invoking_event = constructInvokingEvent(constructConfigItem(self.user_list['Users'][1]['UserName']))
+        lambda_event = build_lambda_configurationchange_event(invoking_event, rule_parameters=rule_param)
+        response = RULE.lambda_handler(lambda_event, {})
+        print(response)
+        resp_expected = []
+        resp_expected.append(build_expected_response('COMPLIANT', self.user_list['Users'][1]['UserName']))
+        assert_successful_evaluation(self, response, resp_expected, 1)
+
+    def test_scenario_6_compliant(self):
+        """Test scenario to test compliant user name
+        Keyword arguments:
+        self -- class ComplianceTest
+        """
+        IAM_CLIENT_MOCK.list_users = MagicMock(return_value=self.user_list)
+        rule_param = '{"pattern":".*power.*"}'
+        invoking_event = constructInvokingEvent(constructConfigItem(self.user_list['Users'][2]['UserName']))
+        lambda_event = build_lambda_configurationchange_event(invoking_event, rule_parameters=rule_param)
+        response = RULE.lambda_handler(lambda_event, {})
+        print(response)
+        resp_expected = []
+        resp_expected.append(build_expected_response('COMPLIANT', self.user_list['Users'][2]['UserName']))
+        assert_successful_evaluation(self, response, resp_expected, 1)
+
+    def test_scenario_7_compliant(self):
+        """Test scenario to test compliant user name
+        Keyword arguments:
+        self -- class ComplianceTest
+        """
+        IAM_CLIENT_MOCK.list_users = MagicMock(return_value=self.user_list)
+        rule_param = '{"pattern":"[aA]dmin"}'
```

Comment:
I think I understand your point, you want to try different pattern to test. They are all under Scenario 4 or 5 though.